x-pack/plugins/elastic_assistant/server/routes/knowledge_base/post_knowledge_base.ts

```diff
@@ -49,46 +45,34 @@ export const postKnowledgeBaseRoute = (
         validate: {
           request: {
             params: buildRouteValidationWithZod(CreateKnowledgeBaseRequestParams),
+            query: buildRouteValidationWithZod(CreateKnowledgeBaseRequestQuery),
           },
         },
       },
-      async (
-        context,
-        request: KibanaRequest<CreateKnowledgeBaseRequestParams>,
-        response
-      ): Promise<IKibanaResponse<CreateKnowledgeBaseResponse>> => {
+      async (context, request, response): Promise<IKibanaResponse<CreateKnowledgeBaseResponse>> => {
         const resp = buildResponse(response);
         const ctx = await context.resolve(['core', 'elasticAssistant', 'licensing']);
         const assistantContext = ctx.elasticAssistant;
-        const logger = ctx.elasticAssistant.logger;
-        const telemetry = assistantContext.telemetry;
-        const elserId = await getElser();
         const core = ctx.core;
-        const esClient = core.elasticsearch.client.asInternalUser;
         const soClient = core.savedObjects.getClient();
+        const kbResource = request.params.resource;
+        const modelIdOverride = request.query.modelId;
```

Comment:
Here's the new `modelId` override that can be provided when setting up the KB that's needed for the API integration testing env. 

Might be a good idea to hide this behind the `v2KnowledgeBaseEnabled` FF, as it will re-initialize the ingest pipeline if set. Let's discuss during the review.